components/eam/src/physics/cam/od_common.F90

```diff
@@ -289,6 +289,7 @@ subroutine oro_drag_interface(state,    cam_in,   sgh,      pbuf,   dtime,  nm,
   !-----------------------------------------------------------------------
 
   ncol=state%ncol
+  allocate(dx(ncol),dy(ncol))
```

Comment:
looks like you haven't added a corresponding deallocate statement?
I think the original approach of using `pcols` is preferable to an allocatable.

Reply:
I am not sure what problem the original code `dx(pcols), dy(pcols)` would cause. Their size (pcols) as declared are always no smaller than `dx(ids:ide)`  (`essenrtially dx(1:ncol)` ). When the array is referenced inside subroutine od2d, they will never go out of bound.

Reply:
I confess I don't understand either but I was getting invalid memory reference errors, when I printed out it seemed ncols > pcols so it seemed possible to go out of bounds, but I can test further.

Reply:
if `ncol>pcols` then we have much bigger problems.

Reply:
Just confirmed that ncol is not greater than pcols, a relief.  Also on my machine allocating dxmeter and dymeter as pcols didn't fix the issue.  I'm trying `dxmeter(its:ite)` now for the declaration

Reply:
Just confirmed that ncol is not greater than pcols, a relief.  Also on my machine allocating dxmeter and dymeter as pcols didn't fix the issue.  I'm trying `dxmeter(its:ite)` now for the declaration

Reply:
Just confirmed that ncol is not greater than pcols, a relief.  Also on my machine allocating dxmeter and dymeter as pcols didn't fix the issue.  I'm trying `dxmeter(its:ite)` now for the declaration

Reply:
Just confirmed that ncol is not greater than pcols, a relief.  Also on my machine allocating dxmeter and dymeter as pcols didn't fix the issue.  I'm trying `dxmeter(its:ite)` now for the declaration

Reply:
Just confirmed that ncol is not greater than pcols, a relief.  Also on my machine allocating dxmeter and dymeter as pcols didn't fix the issue.  I'm trying `dxmeter(its:ite)` now for the declaration

Reply:
Okay I've dug into this a bit more.  The stack trace on failure is on this
```
delx = dxmeter
dely = dymeter
```
`delx` and `dely` are declared `its:ite`.  I tried declaring dxmeter and dymeter with pcols, this failed, but the second dxmeter(its:ite) worked.  I confess I don't understand why this was tripping a failure for the lines above.  The assignment seems to have the same extent in memory. It should be something like delx(1:ncols) = dxmeter(:pcols).  Are there cases when pcols != ncols?

Reply:
thanks for the explanation @whannah1 very helpful.  is the `its:ite` approach okay?  It appears `dx` and `dy` are declared `pcols` but `delx` and `dely` are `ncols`, so it does make sense (I think) that
```
delx = dxmeter
```
could fail when `ncols < pcols`. It's not clear what the best approach is then.

Reply:
I think changing that line to the following should work:
```
delx(its:ite) = dxmeter(its:ite)
```

The way things are called we get `its=1,ite=ncol`, so I think explicitly using `its:ite` everywhere we can should clear up any issues like this.

Reply:
that's a much better idea.  I can make that change.